cmd/cluster/resize/infra_node.go

```diff
@@ -0,0 +1,306 @@
+		err := r.hive.Get(ctx, client.ObjectKey{Namespace: tempMp.Namespace, Name: tempMp.Name}, mp)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return true, nil
+			}
+			return false, err
+		}
+
+		log.Printf("temporary machinepool %s/%s still exists, continuing to wait", tempMp.Namespace, tempMp.Name)
+		return false, nil
+	}); err != nil {
+		return err
+	}
+
+	// Wait for infra node count to return to normal
+	log.Printf("waiting for infra node count to return to: %d", int(*originalMp.Spec.Replicas))
+	if err := wait.PollImmediate(twentySecondIncrement, twentyMinuteTimeout, func() (bool, error) {
+		nodes := &corev1.NodeList{}
+		selector, err := labels.Parse("node-role.kubernetes.io/infra=")
+		if err != nil {
+			return false, err
+		}
+
+		if err := r.client.List(ctx, nodes, &client.ListOptions{LabelSelector: selector}); err != nil {
+			return false, err
+		}
+
+		switch len(nodes.Items) {
+		case int(*originalMp.Spec.Replicas):
+			log.Printf("found %d infra nodes, infra resize complete", len(nodes.Items))
+			return true, nil
+		default:
+			log.Printf("found %d infra nodes, continuing to wait", len(nodes.Items))
+			return false, nil
+		}
+	}); err != nil {
+		return err
+	}
+
+	log.Printf("[REMINDER] follow the cleanup tasks in https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/infranode_resized.json")
+	return nil
+}
+
+func (r *Resize) getInfraMachinePool(ctx context.Context) (*hivev1.MachinePool, error) {
+	ns := &corev1.NamespaceList{}
+	selector, err := labels.Parse(fmt.Sprintf("api.openshift.com/id=%s", r.clusterId))
+	if err != nil {
+		return nil, err
+	}
+
+	if err := r.hive.List(ctx, ns, &client.ListOptions{LabelSelector: selector, Limit: 1}); err != nil {
+		return nil, err
+	}
+	if len(ns.Items) == 0 {
+		return nil, fmt.Errorf("found 0 namespaces with tag: api.openshift.com/id=%s", r.clusterId)
+	}
+
+	log.Printf("found namespace: %s", ns.Items[0].Name)
+
+	mpList := &hivev1.MachinePoolList{}
+	if err := r.hive.List(ctx, mpList, &client.ListOptions{Namespace: ns.Items[0].Name}); err != nil {
+		return nil, err
+	}
+
+	for _, mp := range mpList.Items {
+		mp := mp
+		if mp.Spec.Name == "infra" {
+			log.Printf("found machinepool %s, with machine type %s", mp.Name, mp.Spec.Platform.AWS.InstanceType)
+			return &mp, nil
+		}
+	}
+
+	return nil, fmt.Errorf("did not find the infra machinepool in namespace: %s", ns.Items[0].Name)
+}
+
+func (r *Resize) embiggenMachinePool(mp *hivev1.MachinePool) (*hivev1.MachinePool, error) {
+	embiggen := map[string]string{
+		"m5.xlarge":  "r5.xlarge",
+		"m5.2xlarge": "r5.2xlarge",
+		"r5.xlarge":  "r5.2xlarge",
+		"r5.2xlarge": "r5.4xlarge",
+		"r5.4xlarge": "r5.8xlarge",
+	}
+
+	newMp := &hivev1.MachinePool{}
+	mp.DeepCopyInto(newMp)
+
+	// Unset fields we want to be regenerated
+	newMp.CreationTimestamp = metav1.Time{}
+	newMp.Finalizers = []string{}
+	newMp.ResourceVersion = ""
+	newMp.Generation = 0
+	newMp.SelfLink = ""
```

Comment:
From the docs:
```
	// Deprecated: selfLink is a legacy read-only field that is no longer populated by the system.
	// +optional
```

Do we need it for anything or can we just blow this line away?

Reply:
+1 not needed, so I'm blowing it away in case the old machinepool has this filled out